File: algorithm/Sorting/quick_sort.py
def quick_sort(data_list):
    if len(data_list) <= 1:
        return data_list
    
    left, right = list(), list()
    pivot = data_list[0]
    
    for index in range(1, len(data_list)):
        if pivot > data_list[index]:
            left.append(data_list[index])
        else:
            right.append(data_list[index])
    
    return quick_sort(left) + [pivot] + quick_sort(right)

File: algorithm/Sorting-C/quick_sort/quick_sort.c
#include "quick.h"

void quick_sort(int *arr, int left, int right) {
    int i, j;
    int temp;
    int pivot = arr[left];
    if (left < right) {
        i = left;
        j = right + 1;
        while (i <= j) {
            do
                i++;
            while (arr[i] < pivot);
            do
                j--;
            while (arr[j] > pivot);
            if (i < j) {
                temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            } else
                break;
        }
        temp = arr[j];
        arr[j] = arr[left];
        arr[left] = temp;
        quick_sort(arr, left, j - 1);
        quick_sort(arr, j + 1, right);
    }
}
